framework/include/auxkernels/TagVectorArrayVariableGradientAux.h

```diff
@@ -0,0 +1,33 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "TagAuxBase.h"
+#include "AuxKernel.h"
+
+/**
+ * Couple a tagged vector, and return its array value
+ */
+class TagVectorArrayVariableGradientAux : public TagAuxBase<ArrayAuxKernel>
+{
+public:
+  static InputParameters validParams();
+
+  TagVectorArrayVariableGradientAux(const InputParameters & parameters);
```

Comment:
move this to test objects. I dont think we ll have users of this auxkernel any time soon